### ISSUE TITLE
Q: interested in joining forces?

### ISSUE THREAD

**Issue by kyle-github** [laurium-labs/PLCTag.jl#5]
Hi, author of libplctag here.  I saw your wrapper.   What language is this?  Julia?

Another user proposed setting up a GitHub organization with projects for the core library and for the wrappers.  Would you be interested in joining such a thing?  Each wrapper would be its own separate project with its own committers and owners and license.

There are starting to be many wrappers and it would be nice if users had one place to go and one location to look for new release information.

Best,
Kyle

**Comment by markhalonen** [comment]
Hi Kyle!

> Hi, author of libplctag here. I saw your wrapper. What language is this? Julia?

Yep, we use Julia for some complex motion planning using mathematical optimization and genetic algorithms. 

> Another user proposed setting up a GitHub organization with projects for the core library and for the wrappers. Would you be interested in joining such a thing? Each wrapper would be its own separate project with its own committers and owners and license.

I'm perfectly fine with it, @krrutkow might be interested in the technical details of that transition, specifically how the Julia Tag Bots work with releasing new versions of the PLCTag.jl package if we are not the "owners" of the organization. 

> There are starting to be many wrappers and it would be nice if users had one place to go and one location to look for new release information.

Awesome to hear! We've had minimal issues with implementing libplctag on a system with 300+ tags! So thank you 🙏

**Comment by kyle-github** [comment]
Hi Mark,

I believe that the repos are owned by the teams that get created in the org.   I am still trying to figure it all out, so please bear with me :-)  I am trying to set it up so that each wrapper team has full control of their independent repo.   That would include actions, issues, PR handling etc.

Unfortunately, I have no clue about Julia, so I do not know what the tag bots do.   Do they do some automated packaging?

Glad to hear that the library works well for you.  What is your primary platform?   Windows?  Linux?   Are the pre-build binaries of any use?

Can you share something about what your PLCs control?  I am always interested in finding out what people are doing.

Best,
Kyle

**Comment by bhalonen** [comment]
@kyle-github here is a page from our customer off of this job.  https://www.relinc.com/custom-equipment/ifpi-equipment/non-destructive-testing/

**Comment by bhalonen** [comment]
Though the screenshots are not of the current solution

**Comment by kyle-github** [comment]
Sorry for the delay, I have set up [libplctag as a GitHub organization](https://github.com/libplctag) now and and we have our first wrapper team working on C#!  I can set up a team with both of you in it for the Julia wrapper if that helps.

Note that I am in the process of getting permission to relicense the core library from just LGPL to a dual MPL 2.0/LGPL license.   Many users have noted that modern development tooling and languages are pushing toward single static executables rather than dynamically linked executables.   This is harder to work with if you do not want your application code to not be LPGL.   Making the MPL a choice solves this problem while still keeping fairly strong copyleft protections and rights.

Please let me know what you think.

**Comment by kyle-github** [comment]
The core library has been relicensed as dual MPL 2.0 and LGPL!   This is released as version 2.1.6 on the main site.   The C# team has been set up and is starting to move over that wrapper to get it ready for prime time.   

Let me know if you are interested in joining the GitHub organization.

Best,
Kyle

**Comment by krrutkow** [comment]
@kyle-github That would be great, I can get this package set up there and registered as an official Julia package then.

**Comment by kyle-github** [comment]
Let me work on this.  I am a noob with GitHub organizations.  Please be patient :-D

Glad to have you join the org!

**Comment by bhalonen** [comment]
I moved the package over.

**Comment by kyle-github** [comment]
I cleaned up the old libplctag.jl that I created as a placeholder.

Thanks!

Have you joined the [forum](https://groups.google.com/forum/#!forum/libplctag)?  It would be nice to have an area for free-form discussion between the wrapper writers.   The C# team is discussion how to structure their wrapper.   There is an emerging consensus that it will follow a pattern just like the Julia wrapper: a lower level, call-for-call wrapper around the C library, and a higher level one that is more idiomatic for the language.

Best,
Kyle
